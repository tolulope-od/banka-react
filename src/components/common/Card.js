import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ cardContent }) => {
  return <div className="account-stats-card">{cardContent}</div>;
};

Card.propTypes = {
  cardContent: PropTypes.object.isRequired
};

export default Card;
