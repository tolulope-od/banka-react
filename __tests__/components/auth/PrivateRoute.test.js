import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { PrivateRoute } from '../../../src/components/auth/PrivateRoute';
import Card from '../../../src/components/common/Card';

const shallowSetup = () => {
  const props = {
    component: <Card cardContent={{}} />,
    auth: {
      isAuthenticated: true
    },
    render: () => ({})
  };

  const enzymeWrapper = shallow(<PrivateRoute {...props} />);

  return { props, enzymeWrapper };
};

describe('<PrivateRoute /> Component', () => {
  const { props } = shallowSetup();

  it('renders the private route component', () => {
    const wrapper = shallow(<PrivateRoute {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders a component that requires authentication', () => {
    const wrapper = shallow(<PrivateRoute {...props} />);
    expect(wrapper.find('Route').length).toEqual(1);
  });
});
