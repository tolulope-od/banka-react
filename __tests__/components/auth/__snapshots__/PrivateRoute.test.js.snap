// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivateRoute /> Component renders the private route component 1`] = `
<Route
  render={[Function]}
/>
`;
